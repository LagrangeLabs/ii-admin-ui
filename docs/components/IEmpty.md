---
title: IEmpty组件
---

# IEmpty 组件

空状态时的展示占位图
当目前没有数据时，用于显式的用户提示

Demo:

```tsx
import React from 'react';
import { IEmpty } from 'ii-admin-ui';
import { RobotOutlined } from '@ant-design/icons';

import './IEmpty.css';

export default () => {
  return (
    <div class="card-container">
      <IEmpty type="lose-data1" description={false}></IEmpty>
      <IEmpty type="lose-data2"></IEmpty>
      <IEmpty type="no-file"></IEmpty>
      <IEmpty type="no-form"></IEmpty>
      <IEmpty type="no-mail"></IEmpty>
      <IEmpty type="no-order"></IEmpty>
      <IEmpty type="no-doc">children</IEmpty>
    </div>
  );
};
```

<!-- <API src='../../src/components/IEmpty'> -->

### IEmpty prop 说明

继承[EmptyProps](https://ant.design/components/empty-cn/#API)

| 属性        | 说明                                           | 类型                                                                                          | 默认值                      | 是否必传 | 版本 |
| ----------- | ---------------------------------------------- | --------------------------------------------------------------------------------------------- | --------------------------- | -------- | ---- |
| type        | 空状态类型                                     | 'lose-data1' \| 'lose-data2' \| 'no-file' \| 'no-form' \| 'no-mail' \| 'no-order' \| 'no-doc' |                             | 否       |      |
| description | 自定义描述内容                                 | ReactNode                                                                                     |                             | 否       |      |
| image       | 设置显示图片，为 string 时表示自定义图片地址。 | ReactNode                                                                                     |                             | 否       |      |
| imageStyle  | 图片样式                                       | CSSProperties                                                                                 | { width: 220, height: 220 } | 否       |      |

对应关系
| 属性 | 说明 |
| ---| ---|
|data | 数据丢失 |
| dataLose| 数据丢失 |
| file| 暂无文件 |
| form| 暂无表单 |
| mail| 暂无邮件 |
| order| 暂无订单 |
| source| 暂无资料 |
