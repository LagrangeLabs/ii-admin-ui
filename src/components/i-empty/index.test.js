import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import IEmpty from './index';

describe("Test IEmpty component on the self's props", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should render type correctly', () => {
    const wrapper = mount(<IEmpty type="lose-data1"></IEmpty>);

    expect(wrapper.text()).toBe('数据丢失'); // 判断有没有对应的文本

    expect(wrapper.render()).toMatchSnapshot();

    wrapper.unmount();
  });

  it('should render description correctly', () => {
    const wrapper = mount(
      <IEmpty type="lose-data1" description="测试数据"></IEmpty>,
    );
    const wrapperNo = mount(
      <IEmpty type="lose-data1" description={false}></IEmpty>,
    );

    expect(wrapper.text()).toBe('测试数据'); // 判断有没有对应的文本
    expect(wrapperNo.text()).toBe(''); // 判断有没有对应的文本

    wrapper.unmount();
  });
  it('should render children correctly', () => {
    const wrapper = mount(
      <IEmpty type="lose-data1" description="测试数据">
        childrentest
      </IEmpty>,
    );

    expect(wrapper.text()).toBe('测试数据childrentest'); // 判断有没有对应的文本

    wrapper.unmount();
  });
});
