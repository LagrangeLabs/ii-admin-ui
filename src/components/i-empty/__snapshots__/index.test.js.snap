// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test IEmpty component on the self's props should render type correctly 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "ant-empty",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "ant-empty-image",
          "style": "width: 220px; height: 220px;",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "alt": "数据丢失",
              "src": "noData.png",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "ant-empty-description",
          },
          "children": Array [
            Node {
              "data": "数据丢失",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "ant-empty-description",
        },
        "children": Array [
          Node {
            "data": "数据丢失",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "ant-empty-image",
            "style": "width: 220px; height: 220px;",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "alt": "数据丢失",
                "src": "noData.png",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "src": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
